src/interfaces/IStakingVesting.sol

```diff
@@ -8,6 +8,38 @@ interface IStakingVesting {
    */
   error ErrBonusAlreadySent();
 
+  /**
+   * @dev Error thrown when the bonus amount is out of bound.
+   * @param value The value that is out of bound.
+   * @param min The minimum bound.
+   * @param max The maximum bound.
+   */
+  error ErrOutOfBound(uint256 value, uint256 min, uint256 max);
+
+  /**
+   * @notice Error thrown when blocks are out of order.
+   * @param index The index of the element that caused the error.
+   * @param currBlock The current block number.
+   * @param prvBlock The previous block number.
+   * @dev This error is raised when currBlock is not greater than prvBlock, indicating a sequence violation.
+   */
+  error ErrOutOfOrder(uint256 index, uint256 currBlock, uint256 prvBlock);
+
+  /**
+   * @notice Error thrown when an array is empty but is expected to contain elements
+   * @dev This error is thrown in operations that require a non-empty array
+   */
+  error ErrEmptyArray();
+
+  struct Reward {
+    uint64 startBlock;
```

Comment:
Using uint64 for block numbers is quite risky, since we have a plan to update the block time to 1 second